### ISSUE TITLE
spidermonkey, in depth

### ISSUE THREAD

**Issue by jrmarino** [repology/repology#550]
This report about "spidermonkey" was rejected:

> report: This might be one of those ports with an unofficial (random) version scheme.
Some of the numbering is based on firefox versioning. and I've been told version 1.7.0 is newer than 1.8.5. Seems like a mess. Not sure where version "56" is coming from. Firefox again maybe?

> response: The scheme looks pretty clear to me, it's e.g. 52.x.y as stated on https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/Releases and https://archive.mozilla.org/pub/spidermonkey/. 1.7.x and 1.8.x looks to be ancient: http://ftp.mozilla.org/pub/js/

I asked Michael Reim to clarify, he said:

> Well, as I understand it, the cause of this confusion is that Mozilla used a pretty standard versioning scheme originally but at some point adopted the "Chromium-style" versioning where version numbers really mean nothing but if a version is newer or older. The Spidermonkey package is the JS engine used in Firefox released as a standalone variant (it's also known as "mozjs" or just "js" e.g. in Arch Linux). What doesn't exactly make things easier to grasp is the fact that in the beginning (up to and including version 4, just looked it up) Mozilla used different versions for their Gecko engine, their actual product Firefox - and apperently Spidermonkey (even up to Firefox 10.0). They all had version numbers independent of each other.
> 
> What is Spidermonkey17 in the FPC refers to version 1.7, which would be from Firefox 2.x or so. The port for Spidermonkey185 is version 1.8.5, which shipped with Firefox 4.0. This mess went on up to Spidermonkey 1.8.8 which shipped with FireFox 10.0 (looks like this version was neither packaged by FreeBSD nor by Arch). The next version released by Mozilla is Spidermonkey 17.0 which became Spidermonkey170 in the FPC since... well, Spidermonkey17 was already taken, I guess. From 17.0 on, Spidermonkey also uses the same version as Firefox and is released as standalone for ESR versions of Firefox (i.e. 24, 31, 38, 45 and currently 52). So they use comprehensible versioning scheme now, but in the past before the scheme change, I'd say chaos used to reign (and FreeBSD's port names are witness to this).
>
> I don't think that there is a version 56 of Spidermonkey - Firefox 56 is not ESR and I didn't find such a release in the Mozilla release notes. One thing that we could consider is renaming our port to Spidermonkey17 after all, even if that might be adding to the confusion. The reason is that Polkit officially requires this version (or a previous one). Then Spidermonkey52 could be added, making Repology happy and reporting multiple versions instead of an outdated port. To make this whole mess worse, there are actually other programs (that need to be taken seriously) which depend on other versions of Spidermonkey... One example is CouchDB which uses 1.8.5 according to the FPC or the port gnome-shell which depends on version 24... Spidermonkey24 wouldn't be so bad as a name - but how to call the 1.8.5 one in a somewhat consistent manner? :(

If correct, the bottom line is that there is, or at least was, two different naming schemes for spidermonkey.  Both could be considered valid, so it's not possible to compare to each other.  I don't know what the solution should be, but it seems to me that an outright "reject" on raising the issue is hasty.

**Comment by AMDmi3** [comment]
I still don't see any problems with the versioning scheme. It was 1.7.0, then 1.8.5, then 17.x and the following versions corresponding to Firefox releases. It's monotonic. Am I missing something here?

What I got from this is that 1.7.0 and 1.8.5 should probably be marked as `legacy` as they are required for third party software. Also should merge `mozjs` into `spidermonkey`.

PS. PRs related to rules should now go to https://github.com/repology/repology-rules/issues

**Comment by jrmarino** [comment]
okay, copy on the new repository and issues location.
Do you have a recommended package name for these legacy spidermonkey ports?

**Comment by AMDmi3** [comment]
I don't, and Repology in fact does not care. Think of repository users, not Repology, when choosing a package name.

**Comment by jrmarino** [comment]
Before new ports are introduced, we check repology for trends in naming and description.  What other repositories call the package is definitely factored into the creation of the port.  For one, this avoids false "uniques" and for another, it provides logical naming for users coming from other systems.

I thought actually that this kind of consolidation was one of your goals with repology (and that repology was successful in influencing it)

**Comment by AMDmi3** [comment]
Yes, it's one of the goals, but I don't think Repology can be an absolute truth here (yet), since many target metapackage names were chosen subjectively (by me solely; without much knowledge of affected projects; without consistent policy), so it's currently more of Repology accepting naming suggestions than suggesting itself.

**Comment by AMDmi3** [comment]
Should be fixed now, needed merge and legacy rules were added.

**Comment by jrmarino** [comment]
thanks!
